internal/pkg/util/yamlproto_test.go

```diff
@@ -0,0 +1,213 @@
+												RegexAction: &RegexAction{
+													Pattern: "some_regex",
+													Replace: "a_replacement",
+												},
+											},
+										},
+									},
+								},
+							},
+							{
+								Type: &aggregationv1.ResultPredicate_RequestNodeFragment_{
+									RequestNodeFragment: &RequestNodeFragment{
+										Field: 2,
+										Action: &ResultAction{
+											Action: &aggregationv1.ResultPredicate_ResultAction_RegexAction_{
+												RegexAction: &RegexAction{
+													Pattern: "some_regex_for_node_fragment",
+													Replace: "another_replacement",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		Description: "KeyerConfiguration + RequestTypeMatch + StringFragment",
+		Parameters: []interface{}{
+			"keyer_configuration_request_type_match_string_fragment.yaml",
+			&KeyerConfiguration{
+				Fragments: []*Fragment{
+					{
+						Rules: []*FragmentRule{
+							{
+								Match: &MatchPredicate{
+									Type: &aggregationv1.MatchPredicate_RequestTypeMatch_{
+										RequestTypeMatch: &RequestTypeMatch{
+											Types: []string{
+												"type.googleapis.com/envoy.api.v2.Endpoint",
+												"type.googleapis.com/envoy.api.v2.Listener",
+											},
+										},
+									},
+								},
+								Result: &ResultPredicate{
+									Type: &StringFragment{
+										StringFragment: "abc",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+var negativeTests = []TableEntry{
+	{
+		Description: "typo in fragments definition, instead of fragments we have fragmentss",
+		Parameters: []interface{}{
+			"typo_in_fragments.yaml",
+			&KeyerConfiguration{},
+		},
+	},
+	{
+		Description: "inexistent field in fragment",
+		Parameters: []interface{}{
+			"inexistent_field.yaml",
+			&KeyerConfiguration{},
+		},
+	},
+	{
+		Description: "malformed yaml",
+		Parameters: []interface{}{
+			"invalid.yaml",
+			&KeyerConfiguration{},
+		},
+	},
+	{
+		Description: "empty yaml",
+		Parameters: []interface{}{
+			"empty.yaml",
+			&KeyerConfiguration{},
+		},
+	},
+}
+
+var _ = Describe("Yamlprotoconverter", func() {
+	DescribeTable("should be able to convert from yaml to proto",
+		func(ymlFixtureFilename string, expectedProto proto.Message) {
+			ymlBytes, err := ioutil.ReadFile(fmt.Sprintf("testdata/%s", ymlFixtureFilename))
+			Expect(err).To(BeNil())
+			// Get an empty copy of the expected proto to use as a recipient of the unmarshaling.
+			protoToUnmarshal := proto.Clone(expectedProto)
+			proto.Reset(protoToUnmarshal)
+			err = FromYAMLToProto(string(ymlBytes), protoToUnmarshal)
+			Expect(err).To(BeNil())
+			Expect(proto.Equal(protoToUnmarshal, expectedProto)).To(Equal(true))
+		},
+		positiveTests...)
+
+	DescribeTable("should not be able to convert from yaml to proto",
+		func(yml string, protoToUnmarshal proto.Message) {
```

Comment:
may be this can be removed from param, since its always `KeyerConfiguration`?

Reply:
I'd rather add a few more negative tests for the inner protos, that way when we expand the list of protos we can build on these tests. How does that sound?

Reply:
Sounds good.